airflow/operators/latest_only_operator.py

```diff
@@ -0,0 +1,55 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import datetime
+import logging
+
+from airflow.models import BaseOperator, TaskInstance
+from airflow.utils.state import State
+from airflow import settings
+
+
+class LatestOnlyOperator(BaseOperator):
+    """
+    Allows a workflow to skip tasks that are not running during the most
+    recent schedule interval.
+
+    If the task is run outside of the latest schedule interval, all
+    directly downstream tasks will be skipped.
+    """
+
+    ui_color = '#bb6bd6'  # nyanza
+
+    def execute(self, context):
+        start_date = context['ti'].start_date
+        left_window = context['dag'].following_schedule(context['execution_date'])
+        right_window = context['dag'].following_schedule(left_window)
+        logging.debug(
+            'Checking latest only with left_window: %s right_window: %s '
+            'start_date: %s', left_window, right_window, start_date)
+        if not left_window < start_date < right_window:
+            logging.info('Not latest execution, skipping downstream.')
+            session = settings.Session()
+            for task in context['task'].downstream_list:
+                ti = TaskInstance(
+                    task, execution_date=context['ti'].execution_date)
+                ti.state = State.SKIPPED
+                ti.start_date = datetime.datetime.now()
+                ti.end_date = datetime.datetime.now()
+                session.merge(ti)
+            session.commit()
+            session.close()
+            logging.info('Done.'')
```

Comment:
Yep.. it's causing Landscape.io to show an issue. Our rule is that landscape must be green for merge.